api/src/main/java/javax/enterprise/inject/spi/AfterBeanDiscovery.java

```diff
@@ -107,4 +109,24 @@
      */
     public <T> Iterable<AnnotatedType<T>> getAnnotatedTypes(Class<T> type);
 
+    /**
+     *
+     * Gives access to a {@link BeanBuilder} to help building
+     * a {@link Bean} instance
+     *
+     * @return a {@link BeanBuilder}
+     * @since 2.0
+     */
+    public <T> BeanBuilder<T> beanBuilder();
```

Comment:
I think we should add _"@throws IllegalStateException if called outside of the observer method invocation"_. The same applies to other builder-producing methods, e.g. `ProcessBeanAttributes.beanAttributesBuilder()`.